internal/peer/node/throttle_grpc.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package node
+
+import (
+	"context"
+
+	"github.com/hyperledger/fabric/common/semaphore"
+	"github.com/hyperledger/fabric/core/peer"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+func initGrpcSemaphores(config *peer.Config) map[string]semaphore.Semaphore {
+	semaphores := make(map[string]semaphore.Semaphore)
+	endorserConcurrency := config.LimitsConcurrencyEndorserService
+	deliverConcurrency := config.LimitsConcurrencyDeliverService
+
+	if endorserConcurrency != 0 {
+		semaphores["/protos.Endorser/ProcessProposal"] = semaphore.New(endorserConcurrency)
+	}
+
+	if deliverConcurrency != 0 {
+		deliverSemaphore := semaphore.New(deliverConcurrency)
+		semaphores["/protos.Deliver/Deliver"] = deliverSemaphore
+		semaphores["/protos.Deliver/DeliverFiltered"] = deliverSemaphore
+		semaphores["/protos.Deliver/DeliverWithPrivateData"] = deliverSemaphore
```

Comment:
Given that we are applying limits per service instead of per rpc call, can we use only the service name such as `/protos.Endorser` and `/protos.Deliver` as the key in the semaphore map? This will simply the above lines and make it less fragile whenever a new rpc is being added to a service. At least to preserve backward compatibility, I assume that we won't change the service name itself. 

If we make the above changes, in the interceptor, we can introduce a function called `getServiceName(info.MethodName)` to get the service name alone to do the lookup in the semaphore map.

Also, please add a comment stating that these services are defined in `fabric-protos` and `fabric-protos-go` repositories. It would help improve the readability.

Reply:
I thought about the same, but wasn't sure if it was better to add parsing to method name. Given your comment and Matt's comment, will parse the method name to get rpc name.

Reply:
Agree, code updated.